tox.ini

```diff
@@ -1,6 +1,6 @@
 [tox]
 skipsdist = True
-envlist = upgrade
+envlist = lint, fmt, tests, full_bundle_tests, test_bundle_1.7, test_selenium_1.7, {test_bundle_deployment}-{1.7,1.8}
```

Comment:
Should we do the same for `test_bundle_1.7, test_selenium_1.7,`? Is there a reason why we kept them pinned to 1.7?

Reply:
Hmm regarding `test_bundle_1.7` and `test_selenium_1.7`, there are two reasons I didn't worked on them:
* There is the [full_bundle_tests environment](https://github.com/canonical/bundle-kubeflow/blob/b9d8527b0112eb4f84a238dd3d81f595100f05a2/tox.ini#L46) which runs both of those environments and is actually configurable (can run tests both for [1.7](https://github.com/canonical/bundle-kubeflow/tree/b9d8527b0112eb4f84a238dd3d81f595100f05a2/tests-bundle/1.7) and [1.8](https://github.com/canonical/bundle-kubeflow/tree/b9d8527b0112eb4f84a238dd3d81f595100f05a2/tests-bundle/1.8))
* All those 3 environments have been mainly designed to run in a self-hosted runners. At the moment, `selenium` tests are [failing](https://github.com/canonical/bundle-kubeflow/issues/676) and we do not have the capacity to debug due to not being able to SSH into the runners. At the same time, we have moved towards [using UATs instead](https://github.com/canonical/bundle-kubeflow/pull/739). Thus, `test_bundle_1.7` and `test_selenium_1.7` are not used anywhere in our CI.

On top of this, [tests-bundle](https://github.com/canonical/bundle-kubeflow/tree/b9d8527b0112eb4f84a238dd3d81f595100f05a2/tests-bundle/1.8) which, as I said above, are designed for running in self-hosted environments with microk8s and [tests/integration](https://github.com/canonical/bundle-kubeflow/blob/b9d8527b0112eb4f84a238dd3d81f595100f05a2/tests/integration/test_bundle_deployment.py) which are designed for running in cloud (and are k8s-agnostic), have an overlap in terms of functionality. Ideally, we would want one test environment configurable enough for any environment. 

With all that in mind, I think that we should not configure those two environments at the moment and instead have a separate effort of cleaning/refactoring tests in this repo. (e.g. even removing tests for unsupported versions)

I 'm open to discussing this more. I had in mind to bring this up with the team too but haven't done so yet. Maybe I can have an issue(s) with all this context too.

Reply:
> With all that in mind, I think that we should not configure those two environments at the moment and instead have a separate effort of cleaning/refactoring tests in this repo. (e.g. even removing tests for unsupported versions)

I agree, let's remove them if they are not relevant anymore. A good compromise is to create an issue to track this in the future (:

Reply:
I 'm working on the issue but it requires more effort in order to summarize the exact state and pain points of those tests. Could we move forward with merging and I 'll come back later and link it here once it's ready.